src/store/selectors/solanaWallet.ts

```diff
@@ -207,61 +250,84 @@ export const stakedAccountsArray = createSelector(
+      if (!allCollaterals[assetAddress.toString()]) {
+        return {
+          maxDeposit: new BN(0),
+          decimals: 6
+        }
       }
-    }
 
-    if (allCollaterals[assetAddress.toString()]?.symbol === 'WSOL') {
-      let newBalance = wSOLBalance.sub(
-        new BN(21 *
-          (10 **
-            (allCollaterals[assetAddress.toString()].reserveBalance.scale - 4)
+      if (allCollaterals[assetAddress.toString()]?.symbol === 'WSOL') {
+        let newBalance = wSOLBalance.sub(
+          new BN(21 * 10 ** (allCollaterals[assetAddress.toString()].reserveBalance.scale - 4))
+        )
+        newBalance = newBalance.lt(
+          allCollaterals[assetAddress.toString()].maxCollateral.val.sub(
+            allCollaterals[assetAddress.toString()].reserveBalance.val
           )
         )
-      )
-      newBalance = newBalance.lt(
-        allCollaterals[assetAddress.toString()].maxCollateral.val
-          .sub(allCollaterals[assetAddress.toString()].reserveBalance.val)
-      )
-        ? newBalance
-        : allCollaterals[assetAddress.toString()].maxCollateral.val
-          .sub(allCollaterals[assetAddress.toString()].reserveBalance.val)
+          ? newBalance
+          : allCollaterals[assetAddress.toString()].maxCollateral.val.sub(
+            allCollaterals[assetAddress.toString()].reserveBalance.val
+          )
+        return {
+          maxDeposit: newBalance.lt(new BN(0)) ? new BN(0) : newBalance,
+          decimals: allCollaterals[assetAddress.toString()].reserveBalance.scale
+        }
+      }
+
       return {
-        maxDeposit: newBalance.lt(new BN(0)) ? new BN(0) : newBalance,
-        decimals: allCollaterals[assetAddress.toString()].reserveBalance.scale
+        decimals: assetBalance.decimals,
+        maxDeposit: assetBalance.balance.lt(
+          allCollaterals[assetAddress.toString()].maxCollateral.val.sub(
+            allCollaterals[assetAddress.toString()].reserveBalance.val
+          )
+        )
+          ? assetBalance.balance
+          : allCollaterals[assetAddress.toString()].maxCollateral.val.sub(
+            allCollaterals[assetAddress.toString()].reserveBalance.val
+          )
       }
     }
-
-    return {
-      decimals: assetBalance.decimals,
-      maxDeposit: assetBalance.balance.lt(
-        allCollaterals[assetAddress.toString()].maxCollateral.val
-          .sub(allCollaterals[assetAddress.toString()].reserveBalance.val)
-      )
-        ? assetBalance.balance
-        : allCollaterals[assetAddress.toString()].maxCollateral.val
-          .sub(allCollaterals[assetAddress.toString()].reserveBalance.val)
+  )
+export const maxAvailableDeposit = createSelector(
+  vaultSwap, accounts, (allVaultSwap, tokensAccounts) => {
+    const collateralAccount = tokensAccounts[allVaultSwap.collateral.toString()]
+    if (allVaultSwap.collateral === DEFAULT_PUBLICKEY) {
+      return new BN(0)
+    } else {
+      return collateralAccount.balance || new BN(0)
```

Comment:
i think "??" might be better; also declaration of collateralAccount could be moved here